pkg/agent/plugin/nodeattestor/httppop/httppop.go

```diff
@@ -0,0 +1,203 @@
+package httppop
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"sync"
+
+	"github.com/hashicorp/hcl"
+	nodeattestorv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/plugin/agent/nodeattestor/v1"
+	configv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/service/common/config/v1"
+	"github.com/spiffe/spire/pkg/common/catalog"
+	"github.com/spiffe/spire/pkg/common/plugin/httppop"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	pluginName = "httppop"
+)
+
+func BuiltIn() catalog.BuiltIn {
+	return builtin(New())
+}
+
+func builtin(p *Plugin) catalog.BuiltIn {
+	return catalog.MakeBuiltIn(pluginName,
+		nodeattestorv1.NodeAttestorPluginServer(p),
+		configv1.ConfigServiceServer(p))
+}
+
+type configData struct {
+	port               int
+	agentName          string
+	attestationPayload []byte
+}
+
+type Config struct {
+	HostName       string `hcl:"hostname"`
+	AgentName      string `hcl:"agentname"`
+	Port           int    `hcl:"port"`
+	AdvertisedPort int    `hcl:"advertised_port"`
+}
+
+type Plugin struct {
+	nodeattestorv1.UnsafeNodeAttestorServer
+	configv1.UnsafeConfigServer
+
+	m sync.Mutex
+	c *Config
+}
+
+func New() *Plugin {
+	return &Plugin{}
+}
+
+func (p *Plugin) ServeNonce(agentName string, nonce string) (err error) {
```

Comment:
I think these functions should be unexported?

Reply:
Updated.